linera-base/src/identifiers.rs

```diff
@@ -574,6 +584,92 @@ impl std::str::FromStr for Owner {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename = "AccountOwner")]
+enum SerializableAccountOwner {
+    User(Owner),
+    Application(ApplicationId),
+}
+
+impl Serialize for AccountOwner {
+    fn serialize<S: serde::ser::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_string())
+        } else {
+            match self {
+                AccountOwner::Application(app_id) => SerializableAccountOwner::serialize(
+                    &SerializableAccountOwner::Application(*app_id),
+                    serializer,
+                ),
+                AccountOwner::User(owner) => SerializableAccountOwner::serialize(
+                    &SerializableAccountOwner::User(*owner),
+                    serializer,
+                ),
+            }
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for AccountOwner {
+    fn deserialize<D: serde::de::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        if deserializer.is_human_readable() {
+            let s = String::deserialize(deserializer)?;
+            let value = Self::from_str(&s).map_err(serde::de::Error::custom)?;
+            Ok(value)
+        } else {
+            let value = SerializableAccountOwner::deserialize(deserializer)?;
+            match value {
+                SerializableAccountOwner::Application(app_id) => {
+                    Ok(AccountOwner::Application(app_id))
+                }
+                SerializableAccountOwner::User(owner) => Ok(AccountOwner::User(owner)),
+            }
+        }
+    }
+}
+
+impl Display for AccountOwner {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AccountOwner::User(owner) => {
+                write!(f, "User:{}", owner)?;
+            }
+            AccountOwner::Application(app_id) => {
+                write!(f, "Application:{}", app_id)?;
+            }
+        };
+
+        Ok(())
+    }
+}
+
+impl FromStr for AccountOwner {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Some(owner) = s.strip_prefix("User:") {
+            Ok(AccountOwner::User(
+                Owner::from_str(owner).expect("Getting Owner should not fail"),
```

Comment:
This should return an error, not panic. (Also below.)
It's definitely reachable: Someone just needs to send a string `User:xyz`.

```suggestion
                Owner::from_str(owner).context("Getting Owner should not fail")?,
```

Reply:
Ah right, good point